### PR TITLE
Adds Vector3, Matrix3, Quaternion, RollPitchYaw and api::Rotation bindings

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #11  #14 

## Summary
- FFI Bindings for:
  -  Vector4, Matrix3, Quaternion
- Rust API for:
  -  Vector4, Matrix3, Quaternion

- https://github.com/maliput/maliput-rs/pull/44 was merged into this branch/PR

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: maliput-sys/src/math/math.h [hunk 1]
```diff
@@ -92,5 +89,119 @@ rust::String Vector3_to_str(const Vector3& self) {
...
+}
+
+bool Quaternion_equals(const Quaternion& self, const Quaternion& other) {
+  return self == other;
+}
+
+rust::String Quaternion_to_str(const Quaternion& self) {
+  std::stringstream ss;
+  ss << self;
+  return {ss.str()};
```
Comment (agalbachicar) [inline maliput-sys/src/math/math.h:203]: We may want to add to_str() method to the Quaternion class.

File: maliput-sys/tests/math_test.rs [hunk 2]
```diff
@@ -74,4 +111,319 @@ mod math_test {
...
+    fn matrix3_is_singular() {
+        let m = Matrix3_new(6.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0);
+        assert_eq!(m.is_singular(), m.determinant() == 0.0);
+    }
+
+    #[test]
+    fn matrix3_transpose() {
+        let m = Matrix3_new(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0);
+        let t = Matrix3_transpose(&m);
+        assert_eq!(Matrix3_row(&m, 1).z(), Matrix3_row(&t, 2).y());
```
Comment (agalbachicar) [inline maliput-sys/tests/math_test.rs:232]:
```suggestion
        let m = Matrix3_new(1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0);
        let t = Matrix3_transpose(&m);
        let n = Matrix3_new(1.0, 4.0, 7.0, 2.0, 5.0, 8.0, 3.0, 6.0, 9.0);
        assert_eq!(Matrix3_equals(&t, &n));
```

**Review by agalbachicar — COMMENTED** [review]
Checkpoint, but it's looking good!

**Comment by francocipollone** [comment]
I accidentaly merged https://github.com/maliput/maliput-rs/pull/44 which uses this PR's branch as target. So I will continue as this is.

File: maliput-sys/tests/math_test.rs (hunk 2 above)
Comment (francocipollone) [reply maliput-sys/tests/math_test.rs]: Thanks. Done

File: maliput-sys/src/math/math.h [hunk 3]
```diff
@@ -92,5 +89,119 @@ rust::String Vector3_to_str(const Vector3& self) {
...
+}
+
+rust::String Vector4_to_str(const Vector4& self) {
+  return {self.to_str()};
+}
+
+std::unique_ptr<Matrix3> Matrix3_new(double m00, double m01, double m02,
+                                     double m10, double m11, double m12,
+                                     double m20, double m21, double m22) {
+  return std::make_unique<Matrix3>(std::initializer_list<double>{m00, m01, m02, m10, m11, m12, m20, m21, m22});
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply maliput-sys/src/math/math.h:115]: Done.

**Pull request merged**